comit/src/network/protocols/setup_swap.rs

```diff
@@ -0,0 +1,646 @@
+        match event {
+            RequestResponseEvent::Message {
+                peer,
+                message:
+                    RequestResponseMessage::Request {
+                        request: message, ..
+                    },
+            } => match message {
+                Message::Alice { alice, common, .. } => {
+                    self.bob_receive_hbit_herc20(peer, common, alice)
+                }
+                Message::Bob { bob, common, .. } => {
+                    self.alice_receive_hbit_herc20(peer, common, bob)
+                }
+            },
+            RequestResponseEvent::OutboundFailure { error, .. } => {
+                tracing::warn!("outbound failure: {:?}", error);
+            }
+            RequestResponseEvent::InboundFailure { error, .. } => {
+                tracing::warn!("inbound failure: {:?}", error);
+            }
+            _ => {}
+        }
+    }
+}
+
+impl NetworkBehaviourEventProcess<RequestResponseEvent<Message<Herc20HbitProtocol>, ()>>
+    for SetupSwap
+{
+    fn inject_event(&mut self, event: RequestResponseEvent<Message<Herc20HbitProtocol>, ()>) {
+        match event {
+            RequestResponseEvent::Message {
+                peer,
+                message:
+                    RequestResponseMessage::Request {
+                        request: message, ..
+                    },
+            } => match message {
+                Message::Alice { alice, common, .. } => {
+                    self.bob_receive_herc20_hbit(peer, common, alice)
+                }
+                Message::Bob { bob, common, .. } => {
+                    self.alice_receive_herc20_hbit(peer, common, bob)
+                }
+            },
+            RequestResponseEvent::OutboundFailure { error, .. } => {
+                tracing::warn!("outbound failure: {:?}", error);
+            }
+            RequestResponseEvent::InboundFailure { error, .. } => {
+                tracing::warn!("inbound failure: {:?}", error);
+            }
+            _ => {}
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, Default)]
+pub struct HbitHerc20Protocol;
+
+impl ProtocolName for HbitHerc20Protocol {
+    fn protocol_name(&self) -> &[u8] {
+        b"/comit/setup-swap/hbit-herc20/1.0.0"
+    }
+}
+
+#[derive(Debug, Clone, Copy, Default)]
+pub struct Herc20HbitProtocol;
+
+impl ProtocolName for Herc20HbitProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        b"/comit/setup-swap/herc20-hbit/1.0.0"
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Codec<U: ProtocolName + Send + Clone>(PhantomData<U>);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct CommonParams {
+    pub ethereum_quantity: asset::Erc20Quantity,
+    #[serde(with = "asset::bitcoin::sats_as_string")]
+    pub bitcoin_quantity: asset::Bitcoin,
+    pub ethereum_absolute_expiry: u32,
+    pub bitcoin_absolute_expiry: u32,
+}
+
+impl Display for CommonParams {
```

Comment:
What required `Display` to be implemented? Forwarding to `Debug` is not that useful :)